core/node/libp2p/nat.go

```diff
@@ -0,0 +1,32 @@
+package libp2p
+
+import (
+	"github.com/libp2p/go-libp2p"
+	autonat "github.com/libp2p/go-libp2p-autonat-svc"
+	host "github.com/libp2p/go-libp2p-host"
+	libp2pquic "github.com/libp2p/go-libp2p-quic-transport"
+	"go.uber.org/fx"
+
+	"github.com/ipfs/go-ipfs/core/node/helpers"
+	"github.com/ipfs/go-ipfs/repo"
+)
+
+var NatPortMap = simpleOpt(libp2p.NATPortMap())
+
+func AutoNATService(quic bool) func(repo repo.Repo, mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host) error {
+	return func(repo repo.Repo, mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host) error {
+		// collect private net option in case swarm.key is presented
+		opts, _, err := PNet(repo)
+		if err != nil {
+			// swarm key exists but was failed to decode
+			return err
+		}
+
+		if quic {
+			opts.Opts = append(opts.Opts, libp2p.DefaultTransports, libp2p.Transport(libp2pquic.NewTransport))
+		}
```

Comment:
Small note: Once libp2p starts using fx, this will go away. At that point, we should be able to just pull in the transports group.